### PR TITLE
Create one gRPC Channel per session and transaction

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

We observed that when using 2 parallel threads to open a large number of transactions, response messages from the server would be intermittently dropped, leading to the client stalling. Creating a separate gRPC Channel for each transaction appears to solve the issue.

## What are the changes implemented in this PR?

- Create one gRPC Channel per session
- Create one gRPC Channel per transaction

### CODE REVIEW

**Comment by alexjpwalker** [comment]
Note: Do not delete this branch after merging! It is required in order to demonstrate the issue to the gRPC maintainers.

File: grakn/rpc/transaction.py
Comment (alexjpwalker) [inline grakn/rpc/transaction.py:55]: This is the main fix here: it seems that using a single gRPC Channel was causing a potential issue where responses from the server would be dropped leading to stalling.

Now, we create a separate Channel for each transaction (and session)

File: tests/BUILD
Comment (alexjpwalker) [inline tests/BUILD:55]: This integration test is a reproducible scenario for the error fixed in this PR.

File: tests/integration/test_concurrent.py
Comment (vmax) [inline tests/integration/test_concurrent.py:32]: @alexjpwalker looks like this method doesn't do much, could we remove it safely?

File: tests/integration/test_concurrent.py
Comment (vmax) [inline tests/integration/test_concurrent.py:37]: @alexjpwalker looks like this method doesn't do much, could we remove it safely?

File: tests/integration/test_concurrent.py
Comment (vmax) [inline tests/integration/test_concurrent.py:40]: @alexjpwalker looks like this method doesn't do much, could we remove it safely?

File: tests/integration/test_concurrent.py
Comment (vmax) [inline tests/integration/test_concurrent.py:51]: @alexjpwalker in 2.0 unless you create it, it won't exist

**Review by vmax — APPROVED** [review]

**Pull request merged**
